src/frontend/src/components/ChangeRequestDropdown.tsx

```diff
@@ -0,0 +1,81 @@
+import { isWithinInterval, subDays } from 'date-fns';
+import { Control, Controller } from 'react-hook-form';
+import { AuthenticatedUser, ChangeRequest, wbsPipe } from 'shared';
+import { useAllChangeRequests } from '../hooks/change-requests.hooks';
+import LoadingIndicator from './LoadingIndicator';
+import { useCurrentUser } from '../hooks/users.hooks';
+
+// Filter and sort change requests to display in the dropdown
+const getFilteredChangeRequests = (changeRequests: ChangeRequest[], user: AuthenticatedUser): ChangeRequest[] => {
+  const today = new Date();
+  const fiveDaysAgo = subDays(today, 5);
+
+  const filteredRequests = changeRequests.filter(
+    (cr) => cr.dateReviewed && cr.accepted && isWithinInterval(cr.dateReviewed, { start: fiveDaysAgo, end: today })
+  );
+
+  // The current user's CRs should be at the top
+  filteredRequests.sort((a, b) => {
+    const isSubmitterAUser = a.submitter.userId === user.userId;
+    const isSubmitterBUser = b.submitter.userId === user.userId;
+
+    if (isSubmitterAUser && isSubmitterBUser) return 0;
+    if (isSubmitterAUser) return -1;
+    if (isSubmitterBUser) return 1;
+
+    return a.crId - b.crId;
+  });
+
+  return filteredRequests;
+};
+
+interface ChangeRequestDropdownProps {
+  control: Control<any, any>;
+  name: string;
+}
+
+const ChangeRequestDropdown = ({ control, name }: ChangeRequestDropdownProps) => {
+  const user = useCurrentUser();
+  const { isLoading, data: changeRequests } = useAllChangeRequests();
+  if (isLoading || !changeRequests) return <LoadingIndicator />;
+
+  const filteredRequests = getFilteredChangeRequests(changeRequests, user);
+
+  const approvedChangeRequestOptions = filteredRequests.map((cr) => ({
+    label: `${cr.crId} - ${wbsPipe(cr.wbsNum)} - ${cr.submitter.firstName} ${cr.submitter.lastName} - ${cr.type}`,
+    value: cr.crId
+  }));
+
+  return (
+    <Box sx={{ display: 'flex', justifyContent: 'end' }}>
+      <Controller
+        control={control}
+        name={name}
+        render={({ field: { onChange, value } }) => (
+          <Box sx={{ width: 200, display: 'flex', flexDirection: 'column', justifyContent: 'start' }}>
+            <FormLabel sx={{ alignSelf: 'start' }}>Change Request ID</FormLabel>
+            <Select
+              id="cr-autocomplete"
+              displayEmpty
+              renderValue={(value) => value}
+              value={value}
+              onChange={(event: SelectChangeEvent<number>) => onChange(event.target.value)}
+              size={'small'}
+              placeholder={'Change Request Id'}
+              sx={{ width: '100%', textAlign: 'left' }}
+            >
+              {approvedChangeRequestOptions.map((option) => (
+                <MenuItem key={option.value} value={option.value}>
+                  {option.label}
+                </MenuItem>
+              ))}
+            </Select>
+          </Box>
+        )}
+      />
```

Comment:
I'd do it like this
```suggestion
      <FormControl>
        <FormLabel sx={{ alignSelf: 'start' }}>Change Request ID</FormLabel>
        <Controller
          control={control}
          name={name}
          render={({ field: { onChange, value } }) => (
            <Select
              id="cr-autocomplete"
              displayEmpty
              renderValue={(value) => value}
              value={value}
              onChange={(event: SelectChangeEvent<number>) => onChange(event.target.value)}
              size={'small'}
              placeholder={'Change Request Id'}
              sx={{ width: 200, textAlign: 'left' }}
            >
              {approvedChangeRequestOptions.map((option) => (
                <MenuItem key={option.value} value={option.value}>
                  {option.label}
                </MenuItem>
              ))}
            </Select>
          )}
        />
      </FormControl>
```

Reply:
can you also add this prop to the Select so the dropdown is aligned on the against the right side of the input box: 
```
              MenuProps={{
                anchorOrigin: {
                  vertical: 'bottom',
                  horizontal: 'right'
                },
                transformOrigin: {
                  vertical: 'top',
                  horizontal: 'right'
                }
              }}
```